GBA/Display.zig

```diff
@@ -0,0 +1,122 @@
+const GBA = @import("core.zig").GBA;
+const IO = @import("IO.zig");
+const Color = @import("color.zig").Color;
+
+const VRAM_BASE_ADDR = 0x06000000;
+const SPRITE_VRAM = VRAM_BASE_ADDR + 0x10000;
+const MODE_4_PAGE_SIZE = 0xA000;
+var current_page = VRAM_BASE_ADDR;
+
+/// Controls the capabilities of background layers
+///
+/// Modes 0-2 are tile modes, modes 3-5 are bitmap modes
+pub const DisplayMode = enum(u3) {
+    /// Tiled mode
+    ///
+    /// Provides 4 normal background layers (0-3)
+    Mode0,
+    /// Tiled mode
+    ///
+    /// Provides 2 normal (0, 1) and one affine (2) background layer
+    Mode1,
+    /// Tiled mode
+    ///
+    /// Provides 2 affine (2, 3) background layers
+    Mode2,
+    /// Bitmap mode
+    ///
+    /// Provides a 16bpp full screen bitmap frame
+    Mode3,
+    /// Bitmap mode
+    ///
+    /// Provides two 8bpp (256 color palette) frames
+    Mode4,
+    /// Bitmap mode
+    ///
+    /// Provides two 16bpp 160x128 pixel frames (half size)
+    Mode5,
+};
+
+pub const RefreshState = enum(u1) {
+    Draw,
+    Blank,
+};
+
+pub const DisplayStatus = packed struct(u16) {
+    /// Read only
+    v_refresh: RefreshState,
+    /// Read only
+    h_refresh: RefreshState,
+    /// Read only
+    vcount_triggered: bool = false,
+    enable_vblank_irq: bool = false,
+    enable_hblank_irq: bool = false,
+    enable_vcount_trigger: bool = false,
+    _: u2 = 0,
+    vcount_trigger_at: u8,
+};
+
+pub const ObjCharacterMapping = enum(u1) {
+    /// Tiles are stored in rows of 32 * 64 bytes
+    TwoDimension,
+    /// Tiles are stored sequentially
+    OneDimension,
+};
+
+pub const Visiblity = enum(u1) {
+    Hide,
+    Show,
+};
+
+pub const Priority = enum(u2) {
+    Highest,
+    High,
+    Low,
+    Lowest,
+};
```

Comment:
Felt like this was worth not forgetting that lower is higher, wouldn't hate better names though?

Reply:
That should be fine and convey what you want